azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -156,13 +156,18 @@ func resourceComputeClusterCreate(d *pluginsdk.ResourceData, meta interface{}) e
 		return tf.ImportAsExistsError("azurerm_machine_learning_compute_cluster", *existing.ID)
 	}
 
+	computeClusterAmlComputeProperties := machinelearningservices.AmlComputeProperties{
+		VMSize:        utils.String(d.Get("vm_size").(string)),
+		VMPriority:    machinelearningservices.VMPriority(d.Get("vm_priority").(string)),
+		ScaleSettings: expandScaleSettings(d.Get("scale_settings").([]interface{})),
+	}
+
+	if d.Get("subnet_resource_id") != nil {
+		computeClusterAmlComputeProperties.Subnet = &machinelearningservices.ResourceID{ID: utils.String(d.Get("subnet_resource_id").(string))}
```

Comment:
With `d.GetOk()`, we can check if the property was set, retrieve it and check if it's [not] the zero value

```suggestion
	if subnetId, ok := d.GetOk("subnet_resource_id"); ok && subnetId.(string) != "" {
		computeClusterAmlComputeProperties.Subnet = &machinelearningservices.ResourceID{ID: utils.String(subnetId.(string))}
```